compiler/core/src/javaScript/javaScriptStyles.re

```diff
@@ -369,6 +369,44 @@ let getStylePropertyWithUnits =
   | (_, _) => getStyleProperty(framework, key, colors, value)
   };
 
+let handleNumberOfLines =
+    (framework: JavaScriptOptions.framework, config: Config.t, parameters) =>
+  switch (
+    framework,
+    ParameterMap.find_opt(ParameterKey.NumberOfLines, parameters),
+  ) {
+  | (JavaScriptOptions.ReactDOM, Some(lineCount)) =>
+    open Monads;
+    let lineHeight =
+      switch (
+        ParameterMap.find_opt(ParameterKey.TextStyle, parameters)
+        >>= (
+          (item: Types.lonaValue) =>
+            TextStyle.find(
+              config.textStylesFile.contents.styles,
+              item.data |> Json.Decode.string,
+            )
+        )
+        >>= ((textStyle: TextStyle.t) => textStyle.lineHeight)
+      ) {
+      | Some(lineHeight) => lineHeight
+      | None => 0.0
+      };
+
+    parameters
+    |> ParameterMap.remove(ParameterKey.NumberOfLines)
+    |> ParameterMap.add(
+         ParameterKey.Display,
+         LonaValue.string("inline-block"),
```

Comment:
Should text elements be `block` for now? Thinking about it some more, we don't really support any inline elements yet. So `inline-block` is essentially just `block` but quirkier here. Same goes for other text, even when max-lines case isn't involved. I.e. we could just add `display: block` as a default on Text.


In the future I've been considering having two separate kinds of `Text` layer. One for "block" text that can contain multiple inline elements (e.g. text with different styles, images) and one for "inline" text, that goes within a "block" text (or is auto-wrapped at compile-time).

Reply:
i like the default approach 👍

Reply:
wow, I need my coffee... sigh... I'll add soon :)

Reply:
ok, resolved. I also brought in the default text-align we talked about